### PR TITLE
Dilpreet | Test | Warehouse wise opening balance support in variant

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/28etkva

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 1]
```diff
@@ -1,57 +1,327 @@
...
-
-
-    ngOnInit() {
-        /** This will use for image format */
-        this.imgPath = isElectron ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
+export class StockBalanceComponent implements OnInit, OnDestroy {
+  /** Image path variable */
+  public imgPath: string = '';
+  /** Observable to unsubscribe all the store listeners to avoid memory leaks */
+  private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
```
Comment (the-prateek-mishra) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]: This file is 2 space formatted, make it 4 space formatted. Make sure you have `indent_size` value set to 4 in `.editorconfig` file. Also check the VS Code default formatting (for more info refer [here](https://www.kindacode.com/article/vs-code-how-to-change-indentation-2-spaces-4-spaces/))

**Review by the-prateek-mishra — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts (hunk 1 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]: @prateek-giddh sir this settings is already applied on VS code settings and .editorconfig file

**Review by the-prateek-mishra — APPROVED** [review]

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.html [hunk 2]
```diff
@@ -1,628 +1,262 @@
-<div class="stock-balance-wrapper">
+<giddh-page-loader *ngIf="isLoading"></giddh-page-loader>
+<div class="stock-balance-wrapper" *ngIf="!isLoading">
     <div class="stock-balance-top-bar">
         <div class="stock-balance-top-bar-left-side">
-            <button mat-flat-button class="new-stock-btn">
+            <button mat-flat-button class="new-stock-btn" [routerLink]="'/pages/new-inventory/stock/product/create'">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.html]: make it dynamic for product/service

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.html [hunk 3]
```diff
@@ -1,628 +1,262 @@
...
-                                <p class="stock-list-expanded-content-head">Cloths</p>
-                            </div>
-                        </div>
-                    </div>
-                </div>
+            <div class="group-list-head group-searchbar" *ngIf="showGroupName">
+                <img src="{{ imgPath }}search-new.svg" class="search-icon" />
+                <select-field
+                    [cssClass]="'form-control'"
+                    class=""
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.html]: remove blank class

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 4]
```diff
@@ -1,57 +1,325 @@
...
+		});
+	}
+
+	/**
+	* Get stock varients from stock unqiue name 
+	*
+	* @param {string} uniquename
+	* @memberof StockBalanceComponent
+	*/
+	public getStockVariants(stock: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]: add return type

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 5]
```diff
@@ -1,57 +1,325 @@
...
+		} else {
+			this.allSelectedWarehouse.push(uniqueName);
+		}
+	}
+
+	/**
+	* Callback for selection of group name
+	*
+	* @param {*} field
+	* @memberof CustomFieldsCreateEditComponent
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]:
```suggestion
	* @memberof StockBalanceComponent
```

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 6]
```diff
@@ -1,57 +1,325 @@
...
+	public stockUpdate(stock: any): void {
+		setTimeout(() => {
+			this.inventoryService.updateStock(stock?.stock, stock?.stock?.stockGroup?.uniqueName, stock?.stockUniqueName).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+				if (response?.status === "success") {
+					this.toaster.showSnackBar("success", "Stock updated successfully");
+				} else {
+					this.toaster.showSnackBar("error", response?.message);
+				}
+			});
+		}, 300);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]: change to 3000

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 7]
```diff
@@ -1,57 +1,325 @@
...
+		});
+	}
+
+	/**
+	* Arrange stock groups
+	*
+	* @private
+	* @param {IGroupsWithStocksHierarchyMinItem[]} groups
+	* @param {IOption[]} [parents=[]]
+	* @memberof StockCreateEditComponent
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]:
```suggestion
	* @memberof StockBalanceComponent
```

File: apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts [hunk 8]
```diff
@@ -1,57 +1,325 @@
...
+					this.arrangeStockGroups(group.childStockGroups, parents);
+				}
+			}
+		});
+	}
+
+	/**
+	* Lifecycle hook for destroy
+	*
+	* @memberof StockCreateEditComponent
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts]:
```suggestion
	* @memberof StockBalanceComponent
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 9]
```diff
@@ -711,6 +711,9 @@
                                         <div class="product-option" *ngIf="isVariantAvailable">
                                             <div class="product-option-wrapper">
                                                 <div class="row">
+                                                    <button mat-icon-button class="expand-button">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]: remove this

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 10]
```diff
@@ -166,6 +166,10 @@
             position: relative;
             right: 15px;
         }
+        .expand-button {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: remove this

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**